gef-ui/server/controllers/bank-review-date/get-bank-review-date.ts

```diff
@@ -0,0 +1,87 @@
+import { Response } from 'express';
+import { getDate, getMonth, getYear, parseISO } from 'date-fns';
+import { CustomExpressRequest, isFacilityEndDateEnabledOnGefVersion, parseDealVersion } from '@ukef/dtfs2-common';
+import * as api from '../../services/api';
+import { BankReviewDateViewModel } from '../../types/view-models/bank-review-date-view-model';
+import { asLoggedInUserSession } from '../../utils/express-session';
+import { Facility } from '../../types/facility';
+
+type GetBankReviewDateRequest = CustomExpressRequest<{ params: { dealId: string; facilityId: string }; query: { status: string | undefined } }>;
+
+type HandleGetBankReviewDateParams = {
+  req: GetBankReviewDateRequest;
+  res: Response;
+  previousPage: string;
+};
+
+/**
+ * @param facility - the facility
+ * @param previousPage - the previous page url
+ * @param status - the query parameter status
+ * @returns view model for the bank review date template
+ */
+const getBankReviewDateViewModel = (facility: Facility, previousPage: string, status?: string): BankReviewDateViewModel => {
+  const bankReviewDateViewModel: BankReviewDateViewModel = {
+    dealId: facility.dealId,
+    facilityId: facility._id,
+    previousPage,
+    status,
+  };
+
+  if (typeof facility.bankReviewDate === 'string') {
+    const bankReviewDate = parseISO(facility.bankReviewDate);
+    bankReviewDateViewModel.bankReviewDate = {
+      day: String(getDate(bankReviewDate)),
+      month: String(getMonth(bankReviewDate) + 1),
+      year: String(getYear(bankReviewDate)),
+    };
+  }
+
+  return bankReviewDateViewModel;
+};
+
+/**
+ * Handle get bank review date requests
+ */
+const getBankReviewDate = async ({ req, res, previousPage }: HandleGetBankReviewDateParams) => {
```

Comment:
has this function and the one above been tested?

Reply:
have added more unit tests